c_src/ep_decoder.c

```diff
@@ -896,53 +890,36 @@ fill_default(ErlNifEnv *env, ep_spot_t *spot)
 
             case field_float:
             case field_double:
-                *t++ = state->double_zero;
+                *t++ = enif_make_double(env, 0.0);
                 break;
 
             case field_string:
-            case field_bytes:
-                *t++ = state->binary_nil;
+                *t++ = enif_make_string(env, "", ERL_NIF_LATIN1);
                 break;
 
-            default:
-                *t++ = state->atom_undefined;
+            case field_bytes: {
+                if (!enif_alloc_binary(0, &bin)) {
+                    return RET_ERROR;
+                }
+                *t++ = enif_make_binary(env, &bin);
                 break;
             }
 
-        } else if (field->type == field_enum) {
-            ep_enum_field_t    *efield;
-            efield = field->sub_node->v_fields;
```

Comment:
@vipin211 
investigated and for enums, `v_fields` is a copy of `fields` which is sorted by integer value assigned to the enum. `fields` is sorted arbitrarily by the IDs of the atoms used for the fields. as a result i think our behavior is correct here (default should be the min. enum value) and theirs isn't. i'll take a look at the spec and make a PR upstream.

Reply:
i made some fixes on our local copy to set the enum to the default field, to set bool default value as false etc.
https://github.com/HalloAppInc/enif_protobuf/commit/8848508735de20ea3287f436b5b70595ef307c85

why are we removing this code? dont think we should.
i am not sure about existing tests in this repo - but we should keep this code.

Reply:
checked again and everything's marked as `optional` - i'll make the change to just fill everything with the default value